src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -283,7 +307,31 @@ public void scan() {
 										description = Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.desc", enabledmethod);
 										extraInfo = Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.extrainfo", thirdpartyHost);
 									}
-								} else {
+								} else if (enabledmethod.equals ("TRACE") || enabledmethod.equals ("TRACK")) {
+									log.debug("Verifying a TRACE/TRACK");								
+									MessageAndEvidence mae = testTraceOrTrack(this.getBaseMsg(), enabledmethod);
+									if ( mae != null) {
+										evidence = mae.evidence;
+										alertMessage = mae.message;
+										raiseAlert = true;
```

Comment:
This is not a blocker, but wouldn't it make more sense to set raiseAlert either first or last, and group the setting of all the alert detail variables together?